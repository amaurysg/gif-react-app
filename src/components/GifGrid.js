import React from 'react'
/* import getGifs from '../helpers/getGifs' */
import useFetchGifs from '../hooks/useFetchGifs'
import GifGridItem from './GifGridItem'
/* import { Ring } from 'react-awesome-spinners' */
import PropTypes from 'prop-types'



const GifGrid = ({category}) => {

  //Here Custom Hooks
  //Desestructure object from customs hooks
  const {data:images, loading} = useFetchGifs(category)

  return (
      <>
          <h3>{category}</h3>
          {loading && <p className='animate__animated animate__flash'>Loading...</p> }

          <div className="card-grid" >
        
              {
                //Tomamos array img(tiene la info) y desestrucutamos (id, title, url)
                images.map( ({id, title, url}) => {
                  return (
                 <div key={id}>
                      <GifGridItem 
                        id={id}
                        title={title}
                        url={url}
                        key={id}
                        /> 
                   
                 </div>
                  )
                }
                
                )
              }         
        </div>
      </>
  )
}
/* { ...img }
key={img.id} */

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,

}

export default GifGrid
