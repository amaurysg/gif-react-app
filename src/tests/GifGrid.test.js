import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from "enzyme"
import GifGrid from "../components/GifGrid"
import useFetchGifs from '../hooks/useFetchGifs'

//Fingimos llamada al archivo
jest.mock('../hooks/useFetchGifs')

describe('Test component GifGrid', () => {

  const category = 'Eric Cartman'

  test('Show component correct', () => {

       useFetchGifs.mockReturnValue({
                data: [],
                loading: true,
            });

        const wrapper = shallow(<GifGrid  category={category} />)
        expect(wrapper).toMatchSnapshot()
        
  })

  test('Debe mostrar items cuando se cargan imagenes useFetchGifs', () => {
    
    //Antes de renderizar el componente, simulamos tener data 
    const gifs = [{
            id: 'ABC',
            url: 'https://google.com',
            title: 'ABC Title'
        },
        {
            id: 'ABCss',
            url: 'https://google.cosm',
            title: 'ABCss Title'
        }
      ]
    //Ocupamos useFetchGifs 
      useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
    
    const wrapper = shallow(<GifGrid  category={category} />)
     expect(wrapper).toMatchSnapshot();
     //No deber√≠a de existir este <p> en caso que haya data
     expect(wrapper.find('p').exists()).toBe(false)
     //Evaluamos si existen datas dentro de gifgriditem
     expect(wrapper.find('GifGridItem').length).toBe(gifs.length)

  })
  
  
  
})
