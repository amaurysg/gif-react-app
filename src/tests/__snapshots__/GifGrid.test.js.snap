// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component GifGrid Debe mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3>
    Eric Cartman
  </h3>
  <div
    className="card-grid"
  >
    <div
      key="ABC"
    >
      <GifGridItem
        id="ABC"
        key="ABC"
        title="ABC Title"
        url="https://google.com"
      />
    </div>
    <div
      key="ABCss"
    >
      <GifGridItem
        id="ABCss"
        key="ABCss"
        title="ABCss Title"
        url="https://google.cosm"
      />
    </div>
  </div>
</Fragment>
`;

exports[`Test component GifGrid Show component correct 1`] = `
<Fragment>
  <h3>
    Eric Cartman
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
